### DISCUSSION TITLE
Where can I install this extension?

### DISCUSSION THREAD

**Discussion by NoobTracker** [ScratchAddons/ScratchAddons#145] [Help] [answered]
I searched and wasn't able to find any information on how to install this extension. Do I have to download it? It's not in the Firefox addons.

**Comment by TheSeagullDev** [comment]
Are you using firefox? I'm not sure how to use it with firefox but if you use chrome I can help you.

**Comment by NoobTracker** [comment]
@Thingyman-DEV I'm using Firefox. But I also didn't find any information on how to install it in Chrome.

**Comment by TheSeagullDev** [comment]
I'm going to try it in firefox and see if it works, using the method I used with chrome.

**Comment by NoobTracker** [comment]
... and what's the method you used with chrome?

**Comment by TheSeagullDev** [comment]
I downloaded the folder, went into developer mode, loaded the extension, and switched out of developer mode. However that does not seem to work with firefox.

**Comment by NoobTracker** [comment]
Ah ...
Maybe a userscript version would be neat.

**Comment by TheSeagullDev** [comment]
I tried packing it myself but firefox said it couldn't be verified so it didn't let me use it. I think the devs of this would need to get it signed by firefox [by going here.](https://extensionworkshop.com/documentation/publish/signing-and-distribution-overview/#distributing-your-addon) Unfortunately firefox removed support for unpacked extensions a while ago.

**Comment by NoobTracker** [comment]
Meh ...
Looks like we need a userscript.

**Comment by GrahamSH-LLK** [comment]
> Meh ...
> Looks like we need a userscript.

It won't work as a userscript. It will be on the stores eventually.

**Comment by NoobTracker** [comment]
> 
> 
> > Meh ...
> > Looks like we need a userscript.
> 
> It won't work as a userscript. 

Uh ... why? Because there are settings and stuff? I guess you could just set the userscript to scratch.mit.edu/* and then add a little button somewhere (maybe in the dropdown) for the extension settings.

**Comment by mxmou** [comment]
> > It won't work as a userscript.
> 
> Uh ... why? Because there are settings and stuff? I guess you could just set the userscript to scratch.mit.edu/* and then add a little button somewhere (maybe in the dropdown) for the extension settings.

Because it wasn't designed to be a userscript.

**Comment by NoobTracker** [comment]
Do you think that it could be converted to one?

**Comment by jeffalo** [comment]
technically yes practically no.

Firefox's system for installing addons not on the store is pretty bad to be honest, but the idea is you download this (git clone or just download a zip) unzip it, and then follow this guide 

https://extensionworkshop.com/documentation/develop/debugging/

i don't think there's a better way to do it on Firefox

**Comment by WorldLanguages** [comment]
Nope - definitely can't converted into an userscript.

**Comment by WorldLanguages** [comment]
To install this extension, just clone the repo and import it to your favorite browser as an extension/addon in debug/developer mode :)
This will soon be in stores.

**Comment by jeffalo** [comment, marked as answer]
for chrome try https://www.cnet.com/how-to/how-to-install-chrome-extensions-manually/

**Comment by NoobTracker** [comment]
Ah okay, thank you!

**Comment by NoobTracker** [comment]
Ah now it works. Phew
